ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java

```diff
@@ -1558,7 +1560,10 @@ protected synchronized int End (boolean reportHeuristics)
             {
                 if (heuristicDecision != TwoPhaseOutcome.HEURISTIC_ROLLBACK)
                 {
-                    TxStats.getInstance().incrementCommittedTransactions();
+                    if (startTime == null)
```

Comment:
I don't think startTime can be null here as line 1509 sets this if TxStats.enabled we are in 1159 which also checks .enabled()